spicy/toolchain/src/compiler/codegen/parsers/literals.cc

```diff
@@ -75,6 +75,14 @@ struct Visitor : public visitor::PreOrder {
     void operator()(hilti::ctor::Bytes* n) final {
         auto len = builder()->integer(static_cast<uint64_t>(n->value().size()));
 
+        auto literal = builder()->id(
+            hilti::ID(fmt("__bytes_%" PRIu64, hilti::util::uitoa_n(hilti::util::hash(n->value()), 16, 5))));
+
+        if ( ! pb()->cg()->haveAddedDeclaration(literal->id()) ) {
+            auto d = builder()->constant(literal->id(), builder()->expression(n));
+            pb()->cg()->addDeclaration(d);
+        }
```

Comment:
This feels quite close to the plumbing for string interning without quite getting there, would it make sense to formalize that in an intern pool or something, then just look up declarations in that for this specific case? :)

or a dedicated `symbol` type? But that is out of scope here for sure

obviously I'm lacking some context, not sure if string interning is or has happened (or is happening and I can't find it formally), so let me know if so

Reply:
Admittably, I copied and adapted this from what we already do for regular expressions. Agree that it would make sense to add some infrastructure here, let me try that.

Reply:
Done, @evantypanski have a look.

Reply:
I definitely prefer that, I think interning would be useful if we were comparing constants a lot and had a symbol type, but I don't think that'll ever be a common enough use case, thanks!